docs/developers/quickstart/deploy-smart-contract/hardhat.md

```diff
@@ -0,0 +1,205 @@
+    require(balances[msg.sender] >= amount, "Insufficient balance.");
+    balances[msg.sender] -= amount;
+    balances[recipient] += amount;
+  }
+}
+```
+
+Then, to compile your contract, run `npx hardhat compile`.
+
+## Create a deployment script
+
+To deploy your contract, we'll need to write a deployment script. First, create a folder `scripts`, and replace the `deploy.js` within it. Add the following code:
+
+```javascript
+async function main() {
+  const [deployer] = await ethers.getSigners();
+
+  console.log("Deploying contracts with the account:", deployer.address);
+
+  console.log("Account balance:", (await deployer.getBalance()).toString());
+
+  const Token = await ethers.getContractFactory("Token");
+  const token = await Token.deploy();
+
+  console.log("Token address:", token.address);
+}
+
+main()
+  .then(() => process.exit(0))
+  .catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
+```
+
+## Deploy your contract
+
+You can deploy with Hardhat by [adding ConsenSys zkEVM to your `hardhat.config.js`](#hardhatconfigjs), but you can also use [Truffle Dashboard](#truffle-dashboard) to deploy directly with your MetaMask wallet.
+
+### Truffle Dashboard
+
+[Truffle Dashboard](https://trufflesuite.com/docs/truffle/how-to/use-the-truffle-dashboard/) allows you to forgo saving your private keys locally, instead connecting to your MetaMask wallet for deployments. To deploy with Truffle Dashboard:
+
+1. [Install Truffle using the recommended installation procedure](https://trufflesuite.com/docs/truffle/how-to/install/)
+1. Run `truffle dashboard` in your terminal, which will open a window on port `24012`.
+1. Navigate to `localhost:24012` in your browser. Please ensure that Dashboard is connected to the ConsenSys zkEVM testnet by connecting your MetaMask wallet to ConsenSys zkEVM. For reference, the ConsenSys zkEVM testnet network ID is `59140`.
+
+   ![confirm network](../../../assets/dashboard_network.png)
+
+1. Add the the the Dashboard network to your `hardhat.config.js`
+
+   ```javascript
+   networks: {
+     // other networks
+     truffledashboard: {
+       url: "http://localhost:24012/rpc"
+     }
+   },
+   ```
+
+1. In a separate terminal, in your Hardhat project, run
+
+   ```bash
+   npx hardhat run scripts/deploy.js --network truffledashboard
+   ```
+
+1. Navigate back to `localhost:24012`. You should see a prompt asking your to confirm the deployment. Click **Confirm**.
+
+   ![confirm deployment](../../../assets/dashboard_deploy.png)
+
+1. You should see something similar to:
+
+   ```bash
+   Deploying contracts with the account: YOUR_ACCOUNT_NUMBER
+   Account balance: 71790921294697313
+   Token address: 0xD8b30F2B76361AD82d086414a1c47b0062DCBaf3
+   ```
+
+### `hardhat.config.js`
+
+To deploy to ConsenSys zkEVM, we'll need to add the network to our `hardhat.config.js`. To do this:
+
+1. Create a `.env` file in the `truffle` folder with your wallet's mnemonic and Infura API key.
```

Comment:
The example here uses the private key (not mnemonic)? Also, shouldn't we create the `.env` in the root directory of the project?

Reply:
ah yep